src/LocalDatastore.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+/* global localStorage */
+
+import CoreManager from './CoreManager';
+
+import type ParseObject from './ParseObject';
+
+const DEFAULT_PIN = '_default';
+const PIN_PREFIX = 'parsePin_';
+
+const LocalDatastore = {
+  fromPinWithName(name: string) {
+    const controller = CoreManager.getLocalDatastoreController();
+    return controller.fromPinWithName(name);
+  },
+
+  pinWithName(name: string, value: any) {
+    const controller = CoreManager.getLocalDatastoreController();
+    return controller.pinWithName(name, value);
+  },
+
+  unPinWithName(name: string) {
+    const controller = CoreManager.getLocalDatastoreController();
+    return controller.unPinWithName(name);
+  },
+
+  _getLocalDatastore() {
+    const controller = CoreManager.getLocalDatastoreController();
+    return controller.getLocalDatastore();
+  },
+
+  _clear(): void {
+    var controller = CoreManager.getLocalDatastoreController();
```

Comment:
prefer const

Reply:
I added 'no-var' to eslint rule and there are quite a few there. Should I fix them in this PR?

Reply:
Another one probably :(

Reply:
@dplewis please, in another PR for the linter fixes, now it's a mess of changes ;)